taichi/ir/frontend_ir.h

```diff
@@ -557,6 +557,19 @@ class TensorElementExpression : public Expression {
   }
 };
 
+class GlobalThreadIndexExpression : public Expression {
+ public:
+  Stmt *loop_stmt;
+  GlobalThreadIndexExpression(Stmt *stmt) : loop_stmt(stmt) {
+  }
+
```

Comment:
Could you also add a `type_check` member function (similar to that in `RandExpression`) which simply returns `Primitive::i32`?